Marlin/Version.h

```diff
@@ -45,6 +45,16 @@
 #define STRING_DISTRIBUTION_DATE "2016-04-16 12:00"
 
 /**
+ * Required minimum Configuration.h and Configuration_adv.h file versions.
+ *
+ * You must increment this version number for every significant change such as,
+ * but not limited to: ADD, DELETE RENAME OR REPURPOSE any directive/option on
+ * the configuration files.
+ */
+#define REQUIRED_CONFIGURATION_H_VERSION 1
+#define REQUIRED_CONFIGURATION_ADV_H_VERSION 1
```

Comment:
The problem I see is that we could have multiple config updates during a tag release.. I leave it up to you to decide which is best.